home-work.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import "fmt"
+
+// Task #1
+func factorial(i uint) uint {
+	sum := 1
+	for z := 1; z <= int(i); z++ {
+		sum *= z
+	}
+	return uint(sum)
+}
+
+// Task #2
+type Point struct {
+	x, y int
+}
+
+type Square struct {
+	start Point
+	a     uint
+}
+
+func (s Square) End() Point {
+	return Point{int(s.a), int(s.a)}
```

Comment:
Not sure that these values are correct.
I guess `End` has `s.start.x + a`, `s.start.y - a` coordinates.